### PR TITLE
bulk frontend

### PR DESCRIPTION
**Pull request by benauriemma**

### CODE REVIEW

**Comment by benauriemma** [comment]
Changes pulled from bauriemma-bulk here

File: rackcity/api/serializers/asset_serializers.py [hunk 1]
```diff
@@ -188,6 +188,8 @@ def normalize_bulk_asset_data(bulk_asset_data):
     del bulk_asset_data['power_port_connection_2']
     if not bulk_asset_data['asset_number']:
         del bulk_asset_data['asset_number']
+    if not bulk_asset_data['hostname']:
+        del bulk_asset_data['hostname']
```
Comment (juliasaveliff) [inline rackcity/api/serializers/asset_serializers.py]: 👍

File: rackcity/views/asset_views.py [hunk 2]
```diff
@@ -760,7 +765,7 @@ def asset_bulk_upload(request):
                     status=HTTPStatus.BAD_REQUEST
                 )
         # Check that all hostnames in file are case insensitive unique
-        if asset_data['hostname']:
+        if 'hostname' in asset_data and asset_data['hostname']:
```
Comment (juliasaveliff) [inline rackcity/views/asset_views.py]: 👍

File: rackcity/views/asset_views.py [hunk 3]
```diff
@@ -1006,8 +1012,12 @@ def network_bulk_upload(request):
...
             status=HTTPStatus.BAD_REQUEST
         )
-    csv_string = StringIO(data['import_csv'])
-    csvReader = csv.DictReader(csv_string)
+    base_64_csv = data['import_csv']
+    csv_bytes_io = BytesIO(
+        b64decode(re.sub(".*base64,", '', base_64_csv))
+    )
+    csv_string_io = StringIO(csv_bytes_io.read().decode('UTF-8'))
+    csvReader = csv.DictReader(csv_string_io)
```
Comment (juliasaveliff) [inline rackcity/views/asset_views.py]: 👍

File: rackcity/views/it_model_views.py [hunk 4]
```diff
@@ -390,11 +392,15 @@ def model_bulk_upload(request):
     data = JSONParser().parse(request)
     if 'import_csv' not in data:
         return JsonResponse(
-            {"failure_message": "Bulk upload request should have a parameter 'import_csv'"},
+            {"failure_message": "Bulk upload request should have a parameter 'file'"},
             status=HTTPStatus.BAD_REQUEST
         )
```
Comment (juliasaveliff) [inline rackcity/views/it_model_views.py]: note to self @juliasaveliff you will need to update this error message (might cause merge conflicts in my PR)

File: rackcity/views/rackcity_utils.py [hunk 5]
```diff
@@ -156,11 +158,15 @@ def int_string_comparison(existing_value, new_value):
...
-def empty_string_null_comparison(existing_value, new_value):
+def empty_vs_null_comparison(existing_value, new_value):
     return (
         (
             isinstance(existing_value, str)
             or isinstance(new_value, str)
+            or isinstance(existing_value, list)
+            or isinstance(new_value, list)
+            or isinstance(existing_value, dict)
+            or isinstance(new_value, dict)
```
Comment (juliasaveliff) [inline rackcity/views/rackcity_utils.py]: mood

File: src/components/elementView/elementTab.tsx [hunk 6]
```diff
@@ -294,32 +294,36 @@ class ElementTab extends React.Component<ElementTabProps, ElementViewState> {
...
+                  this.props.history.push("/bulk-upload/" +
+                    (this.props.element === ElementType.MODEL ? "models" : "assets")
+                  )
+                }}
+                className="add"
+                icon="export"
+                text="Add from CSV file"
+                minimal
+              />
+            ) : null}
```
Comment (juliasaveliff) [inline src/components/elementView/elementTab.tsx]: 👍

**Pull request merged**